Readability.js

```diff
@@ -119,6 +119,36 @@ Readability.prototype = {
   },
 
   /**
+   * Iterate over a NodeList, which doesn't natively fully implement the Array
+   * interface.
```

Comment:
Readability's code needs to run on iOS, so I suspect what you suggest is not an option (sadly).